### PR TITLE
permission check bugfix

### PR DESCRIPTION
**Pull request by ttoomey**
## _Merging this PR_
- use the squash-merge strategy for PRs targeting a release-X branch

## Description
[//]: # (Summarize changes and include links related issue. List any new dependencies)

## Type of change
Bug fix, tests, small cleanups and param checks

references:
[#6817](https://github.com/open-path/Green-River/issues/6817#issuecomment-2486672597)
[Sentry issue](https://green-river.sentry.io/issues/6076011109/events/387c241511ab4c64a79972692181f5bf/) 

## Checklist before requesting review
[//]: # (Remove any items that are not applicable)
- [x] I have performed a self-review of my code
- [x] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [x] I have added spec tests (or not applicable)

### CODE REVIEW

File: app/models/concerns/sql_helper.rb [hunk 1]
```diff
@@ -7,8 +7,6 @@
 # Module for SQL-related helper methods. Assumes a PostgreSQL database.
 # Perhaps there's a way to do this with the active_record_extended gem but I couldn't find it
 module SqlHelper
-  extend ActiveSupport::Concern
```
Comment (ttoomey) [inline app/models/concerns/sql_helper.rb:10]: we don't use SqlHelper as a mixin

File: app/models/grda_warehouse/auth_policies/user_legacy_context.rb [hunk 2]
```diff
@@ -71,7 +71,8 @@ def project_access_group_ids(project_id)
     coc_codes = GrdaWarehouse::Hud::ProjectCoc.
       joins(:project).
       where(p_t[:id].eq(project_id)).
-      pluck(:coc_code)
+      pluck(:coc_code).
+      compact_blank
```
Comment (ttoomey) [inline app/models/grda_warehouse/auth_policies/user_legacy_context.rb:75]: this was the actual bug

File: spec/models/sql_helper_spec.rb [hunk 3]
```diff
@@ -0,0 +1,89 @@
+require 'rails_helper'
+
+RSpec.describe SqlHelper do
```
Comment (ttoomey) [inline spec/models/sql_helper_spec.rb:3]: add missing tests

File: drivers/hopwa_caper/app/models/hopwa_caper/enrollment.rb [hunk 4]
```diff
@@ -6,7 +6,6 @@
 
 module HopwaCaper
   class Enrollment < GrdaWarehouseBase
-    include SqlHelper
```
Comment (ttoomey) [inline drivers/hopwa_caper/app/models/hopwa_caper/enrollment.rb:9]: not used afaict

**Comment by ttoomey** [comment]
deployed to qa

**Review by eanders — APPROVED** [review]
Thanks @ttoomey!

File: app/models/grda_warehouse/auth_policies/project_policy.rb [hunk 5]
```diff
@@ -8,7 +8,7 @@ class GrdaWarehouse::AuthPolicies::ProjectPolicy < GrdaWarehouse::AuthPolicies::
   # expose role permissions. Optionally rename the permission
   [
     [:can_edit_projects, :can_edit?],
-    [:can_delete_projects, :can_delete?],
+    [:can_delete_projects, :can_destroy?],
```
Comment (ttoomey) [inline app/models/grda_warehouse/auth_policies/project_policy.rb:11]: bug fix, standardize on `can_destroy` which seems more intuitive to me; perhaps that's subjective

File: app/views/data_sources/_source_details.haml [hunk 6]
```diff
@@ -8,7 +8,7 @@
       %td
         - if data_span = @data_source.data_span
           = "#{data_span[:start_date]} - #{data_span[:end_date]}"
-    - if (!@data_source.authoritative? || @data_source.hmis?) && can_view_imports? && @data_source.uploads.any? && @data_source.directly_viewable_by?(current_user, permission: :can_view_imports)
+    - if ((!@data_source.authoritative? || @data_source.hmis?) && can_view_imports? && @data_source.uploads.any? && @data_source.directly_viewable_by?(current_user, permission: :can_view_imports)) && can_upload_hud_zips?
```
Comment (ttoomey) [inline app/views/data_sources/_source_details.haml:11]: bug fix: the `UploadsController` checks `require_can_upload_hud_zips` - mirror that logic here. Note this condition should probably be moved into the data source policy in future work

**Review by eanders — APPROVED** [review]

**Pull request merged**